src/core/hub/Hub.sol

```diff
@@ -323,6 +328,8 @@ contract Hub is BatchedMulticall, Auth, Recoverable, IHub, IHubRequestManagerCal
         _isManager(poolId);
 
         shareClassManager.updateSharePrice(poolId, scId, pricePoolPerShare);
+
+        if (address(feeHook) != address(0)) feeHook.accrue(poolId, scId);
```

Comment:
Can we take advance that EVM success when a method doesn't exist for some contract and avoid the `if`?

Reply:
It is not about whether the method exist, but whether the fee hook has been set.

Reply:
Yeah, but the method would not exists for `address(0)`.

Q: I understand this condition exists because we still does't know the implementation, right? But the idea is to always have this implemented along with the Hub

Reply:
Just an idea, what about uploading an empty impl, so that later we can change it for the real one?

Reply:
I think having the if statement is cleaner.